jwst/srctype/srctype.py

```diff
@@ -98,7 +110,29 @@ def set_source_type(input_model):
             else:
                 src_type = 'POINT'
 
-            log.info(f'Input source type is unknown; setting default SRCTYPE = {src_type}')
+            log.info(f'Input source type is unknown; setting default SRCTYPE \
+                     = {src_type}')
+
+        # Check to see if the user specified a source type
```

Comment:
Looking over the whole new logic flow in here, I don't think it's desirable to have the old logic execute first, along with log messages saying that it's setting the source type to this or that because of this or that, only to then have those settings get overridden at the very end by the user-supplied value. I think this whole new if-block should move up to the top of this branch that deals with the single-source modes (i.e. line 77) and have it execute if the user supplied a value and then skip the rest of the existing logic that sets a value based on background target, nodded dither pattern, etc. All of those should be skipped in favor of the user-supplied value, when there is one. So  line 78 turns into an "elif bkg_target:", causing all of those remaining elif blocks to only get executed if there wasn't a user-supplied value.

Reply:
I agree that it makes more sense to have the user-input check at the start, as you describe.  The reason I added my code at the end was because I thought it was needed to avoid duplication of  the slit-handling code - I since realize that the code does not need to be duplicated.  I'm now moving the code to the start of the single-source modes, and the slit-handling still works as desired.   I've tested it on several modes and it works as it should, so I will be including it.